src/definitions/modules/slider.js

```diff
@@ -205,28 +205,26 @@ $.fn.slider = function(parameters) {
             });
           },
           autoLabel: function() {
-            if(module.get.step() != 0) {
-              $labels = $module.find('.labels');
-              if($labels.length != 0) {
-                $labels.empty();
-              }
-              else {
-                $labels = $module.append('<ul class="auto labels"></ul>').find('.labels');
-              }
-              for(var i = 0, len = module.get.numLabels(); i <= len; i++) {
-                var
-                  labelText = module.get.label(i),
-                  $label = (labelText !== "") 
-                    ? !(i % module.get.gapRatio())
-                      ? $('<li class="label">' + labelText + '</li>') 
-                      : $('<li class="halftick label"></li>')
-                    : null,
-                  ratio  = i / len
-                ;
-                if($label) {
-                  module.update.labelPosition(ratio, $label);
-                  $labels.append($label);
-                }
+            $labels = $module.find('.labels');
+            if($labels.length != 0) {
+              $labels.empty();
+            }
+            else {
+              $labels = $module.append('<ul class="auto labels"></ul>').find('.labels');
+            }
+            for(var i = 0, len = module.get.numLabels(); i <= len; i++) {
+              var
+                labelText = module.get.label(i),
+                $label = (labelText !== "")
+                  ? !(i % module.get.gapRatio())
+                    ? $('<li class="label">' + labelText + '</li>')
+                    : $('<li class="halftick label"></li>')
+                  : null,
+                ratio  = i / len
+              ;
+              if($label) {
+                module.update.labelPosition(ratio, $label);
+                $labels.append($label);
```

Comment:
The `for` loop shouldn't execute inside `else` block since it needs to wrap inside the `labels` container?

Reply:
Else block is only creating `auto labels` ul, if block is only asuming you have already created, no?
So if it goes to if statement it will only empty all labels and no one will be added, we could add auto class to `.labels` element.

Reply:
Also when `autoLabel` is called, and if `labels` with no `auto` class is found, it goes to `customLabel`.
So in `autoLabel` if `labels` is found it should already have `auto` class.

Reply:
You're right. I misread the `$labels.empty()` as it remove the `ul` element and thus the for loop would be failed in some situations.